advanced_survey/migrations/0001_initial.py

```diff
@@ -0,0 +1,56 @@
+# Generated by Django 3.2.5 on 2021-07-08 19:16
+
+from django.conf import settings
+from django.db import migrations, models
+import django.db.models.deletion
+
+
+class Migration(migrations.Migration):
+
+    initial = True
+
+    dependencies = [
+        migrations.swappable_dependency(settings.AUTH_USER_MODEL),
+    ]
+
+    operations = [
+        migrations.CreateModel(
+            name='Survey',
+            fields=[
+                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('name', models.CharField(max_length=255)),
+                ('description', models.TextField(blank=True, null=True)),
+                ('created_at', models.DateTimeField(auto_now_add=True)),
+                ('expire_at', models.DateTimeField(blank=True, null=True)),
+            ],
+        ),
+        migrations.CreateModel(
+            name='Question',
+            fields=[
+                ('id', models.BigAutoField(auto_created=True, primary_key=True, serialize=False, verbose_name='ID')),
+                ('page', models.IntegerField(db_index=True, default=0)),
+                ('list_order', models.IntegerField(db_index=True, default=0)),
+                ('question', models.TextField()),
+                ('comment', models.TextField(blank=True, null=True)),
+                ('field_type', models.CharField(choices=[('LineEdit', 'Line Edit'), ('TextArea', 'Text Area'), ('Select', 'Select'), ('Checkbox', 'Checkbix'), ('Radiobox', 'Radiobox'), ('Email', 'Email'), ('URL', 'URL'), ('Phone', 'Phone'), ('Boolean', 'Boolean'), ('Numeric', 'Numeric'), ('File', 'File'), ('Image', 'Image'), ('Date', 'Date'), ('Time', 'Time'), ('Rating', 'Rating')], max_length=255)),
+                ('options_url', models.URLField(blank=True, null=True)),
+                ('options', models.TextField(blank=True, null=True)),
+                ('is_required', models.CharField(blank=True, max_length=255, null=True)),
+                ('is_visible', models.CharField(blank=True, max_length=255, null=True)),
+                ('validation_rules', models.CharField(blank=True, max_length=255, null=True)),
```

Comment:
Could this be a regex? If so, a longer field length might be better

Reply:
edited